src/PerformanceTests/Tests/Enabled/CSharp_CA2252.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.CSharp.Analyzers.Runtime;
+using PerformanceTests.Utilities;
+using PerfUtilities;
+
+namespace CSharpPerformanceTests.Enabled
+{
+    public class CSharp_CA2255
+    {
+        [IterationSetup]
+        public static void CreateEnvironmentCA2252()
+        {
+            var sources = new List<(string name, string content)>();
+            for (var i = 0; i < Constants.Number_Of_Code_Files; i++)
+            {
+                var name = "TypeName" + i;
+                sources.Add((name, @$"
+using System;
+using System.Runtime.Versioning;
+public class {name}
+{{
+    public bool GenericMethod<T>()
+        where T : {name}_Foo, ICloneable
+    {{
+        return true;
+    }}
+}}
+
+[RequiresPreviewFeatures(""Lib is in preview."", Url = ""https://aka.ms/aspnet/kestrel/http3reqs"")]
+public class {name}_Foo
+{{
+}}
```

Comment:
I may extend it to a code that reports multiple different rules in a follow-up PR.